src/vis_cpu/conversions.py

```diff
@@ -22,3 +25,149 @@ def lm_to_az_za(el, m):
     az = -np.arctan2(m, el)
     za = np.pi / 2.0 - np.arcsin(n)
     return az, za
+
+
+def point_source_crd_eq(ra, dec):
+    """
+    Calculate approximate coordinate transform of source locations from 
+    equatorial to Cartesian.
+    
+    Parameters
+    ----------
+    ra, dec : array_like
+        1D arrays of source positions in equatorial coordinates (radians).
+    
+    Returns
+    -------
+    array_like
+        Equatorial coordinates of Healpix pixels, in Cartesian
+        system. Shape=(3, NPIX).
+    """
+    return np.asarray([np.cos(ra)*np.cos(dec), np.cos(dec)*np.sin(ra),
+                     np.sin(dec)])
+
+
+def uvbeam_to_lm(uvbeam, freqs, n_pix_lm=63, polarized=False, **kwargs):
+    """
+    Convert a UVbeam to a uniform (l,m) grid
+
+    Parameters
+    ----------
+    uvbeam : UVBeam object
+        Beam to convert to an (l, m) grid.
+    
+    freqs : array_like
+        Frequencies to interpolate to in [Hz]. Shape=(NFREQS,).
+    
+    n_npix_lm : int, optional
+        Number of pixels for each side of the beam grid. Default is 63.
+    
+    polarized : bool, optional
+        Whether to return full polarized beam information or not. 
+        Default: False.
+    
+    Returns
+    -------
+    ndarray
+        The beam map cube. Shape: (NFREQS, BEAM_PIX, BEAM_PIX) if 
+        `polarized=False` or (NAXES, NFEEDS, NFREQS, BEAM_PIX, BEAM_PIX) if 
+        `polarized=True`.
+    """
+    # Define angle cosines
+    L = np.linspace(-1, 1, n_pix_lm, dtype=np.float32)
+    L, m = np.meshgrid(L, L)
+    L = L.flatten()
+    m = m.flatten()
+    
+    # Apply horizon cut
+    lsqr = L**2. + m**2.
+    n = np.where(lsqr < 1., np.sqrt(1. - lsqr), 0.)
+    
+    # Calculate azimuth and zenith angle
+    az = -np.arctan2(m, L)
+    za = np.pi/2. - np.arcsin(n)
+    
+    # Interpolate beam onto cube
+    efield_beam = uvbeam.interp(az, za, freqs, **kwargs)[0]
+    if polarized:
+        bm = efield_beam[:,0,:,:,:] # spw=0 
+    else:
+        bm = efield_beam[0,0,1,:,:] # (phi, e) == 'xx' component
+    
+    # Peak normalization and reshape output
+    if polarized:
+        Naxes = bm.shape[0] # polarization vector axes
+        Nfeeds = bm.shape[1] # polarized feeds
+        
+        # Separately normalize each polarization channel
+        for i in range(Naxes):
+            for j in range(Nfeeds):
+                if np.max(bm[i,j]) > 0.:
+                    bm /= np.max(bm[i,j])
+        return bm.reshape((Naxes, Nfeeds, len(freqs), n_pix_lm, n_pix_lm))
+    else:
+        # Normalize single polarization channel
+        if np.max(bm) > 0.:
+            bm /= np.max(bm)
+        return bm.reshape((len(freqs), n_pix_lm, n_pix_lm))
+
+
+def eq2top_m(ha, dec):
```

Comment:
I'd be very happy for this function to be renamed now that it's out of the main code... maybe `get_equatorial_to_topo_matrix`?

Reply:
This will be addressed in another active PR.